src/java.base/share/classes/java/lang/reflect/Method.java

```diff
@@ -230,13 +230,16 @@ public int getModifiers() {
 
     /* package */
     Optional<?> setCodeModelIfNeeded(Function<Method, Optional<?>> modelFactory) {
+        if (root != null) {
```

Comment:
This logic is subtly different to the one I pointed out -- this always recurse if the root is non null. The one I pointed out calls the method on the root, but than saves the result on the copy locally. I'm not sure which is better -- but it seems odd to use slightly different idioms in the same class to do effectively the same thing?

Reply:
I applied the pattern you pointed out in the third commit. In this version I intentionally wanted to have the model stored in the root method, this way we only have one reference to the model.